src/node.cc

```diff
@@ -446,22 +453,50 @@ void LoadEnvironment(Environment* env) {
   USE(StartMainThreadExecution(env));
 }
 
+#if defined __linux__ || defined __APPLE__
+static std::atomic<void (*)(int signo, siginfo_t* info, void* ucontext)>
+    previous_sigsegv_action;
+
+void TrapWebAssemblyOrContinue(int signo, siginfo_t* info, void* ucontext) {
+  if (!v8::TryHandleWebAssemblyTrapPosix(signo, info, ucontext)) {
+    CHECK(previous_sigsegv_action.is_lock_free());
+    auto prev = previous_sigsegv_action.load();
+    if (prev != nullptr) {
+      prev(signo, info, ucontext);
+    } else {
+      uv_tty_reset_mode();
+      raise(signo);
+    }
+  }
+}
+#endif  // defined __linux__ || defined __APPLE__
 
 #ifdef __POSIX__
+
 void RegisterSignalHandler(int signal,
-                           void (*handler)(int signal),
-                           bool reset_handler) {
+                           void (*handler)(int signal,
+                                           siginfo_t* info,
+                                           void* ucontext),
```

Comment:
Not a blocker in the slightest but you could fit this on one line and it'd arguably be a little easier to read:
```c++
                           void (*handler)(int, siginfo_t*, void*),
```